### PR TITLE
(PUP-10222) Add test case for looking up class params via hiera data

### PR DESCRIPTION
**Pull request by Magisus**

### CODE REVIEW

**Comment by Magisus** [comment]
This currently causes breakages, which hopefully will ultimately be fixed by https://github.com/puppetlabs/puppet/pull/7917

**Comment by dinhthang5** [comment]
#7917

File: ext/thread_test/environments/production/modules/threader/manifests/params_class.pp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class threader::params_class (String $input = 'softserve') {
```
Comment (mwaggett) [inline ext/thread_test/environments/production/modules/threader/manifests/params_class.pp]: Why does this have a default value but the other one doesn't?

Comment (Magisus) [reply ext/thread_test/environments/production/modules/threader/manifests/params_class.pp]: Oh, I was testing something. But I don't mind having them be different, actually. We prefer the data in hiera in both cases, maybe I should remove the hiera data from the one with the default to test that failed lookup and falling through behavior 🤔

Comment (mwaggett) [reply ext/thread_test/environments/production/modules/threader/manifests/params_class.pp]: yeah, that's what I thought you were going for at first

**Comment by Magisus** [comment]
The submodule update containing the code that fixes this test case just made it into this branch. So I think this can be merged when we're happy with it.

**Pull request merged**